plugins/package-managers/pub/src/main/kotlin/utils/PubCacheReader.kt

```diff
@@ -85,13 +85,13 @@ internal class PubCacheReader(flutterHome: File? = null) {
         val path = if (type == "hosted" && url.isNotEmpty()) {
             // Packages with source set to "hosted" and "url" key in description set to "https://pub.dartlang.org".
             // The path should be resolved to "hosted/pub.dartlang.org/packageName-packageVersion".
-            "hosted/${url.replace("https://", "")}/$packageName-$packageVersion"
+            "hosted/${url.replace("https://", "").replace("/", "%47")}/$packageName-$packageVersion"
```

Comment:
Is "/" really the only character that needs to be escaped here? If there's a chance that other characters might need to be escaped as well, should we use our `String.fileSystemEncode()` function instead?

Reply:
Let's see... https://github.com/dart-lang/pub/blob/ea4a1c854690d3abceb92c8cc2c6454470f9d5a7/doc/cache_layout.md?plain=1#L72 says

> The url of the repository is encoded to a directory name with a weird URI-like
> encoding. This is a mistake that seems costly to fix, but is worth being aware
> of.

And the code in https://github.com/dart-lang/pub/blob/ea4a1c854690d3abceb92c8cc2c6454470f9d5a7/lib/src/source/hosted.dart#L1911
```
  return replace(
    url,
    RegExp(r'[<>:"\\/|?*%]'),
    (match) => '%${match[0]!.codeUnitAt(0)}',
  );
```

`String.fileSystemEncode()` does not work because Pub uses decimal value instead of hex value.

But now that I found the original code, I will add replace function for the missing 9 chars.

Reply:
> Pub uses decimal value instead of hex value.

Wow, that's indeed weird!